### ISSUE TITLE
incompatible error for QPUSubproblemAutoEmbeddingSampler 

### ISSUE THREAD

**Issue by evanlai33** [dwavesystems/dwave-hybrid#76]
The runnable branch with QPUSubproblemAutoEmbeddingSampler may raise error with message: ValueError: Problem graph incompatible with solver. Here's the sample code:

```
# just a random bqm
bqm = dimod.BinaryQuadraticModel({'x': 0.0, 'y': 0.0, 'z': 8.0, 'a': 2.0, 'b': 0.0, 'c': 6.0},
          {('y', 'x'): 2.0, ('z', 'x'): -4.0, ('z', 'y'): -4.0,
          ('b', 'a'): 2.0, ('c', 'a'): -4.0, ('c', 'b'): -4.0, ('a', 'z'): -4.0},
          -1.0, 'BINARY') 

qpusampler = DWaveSampler()

runnable = EnergyImpactDecomposer(max_size=1) | QPUSubproblemAutoEmbeddingSampler(num_reads=1000, qpu_sampler = qpusampler) | SplatComposer()                                                                                  
main = SimpleIterator(runnable, max_iter=10, convergence=3)

init_state = State.from_sample(min_sample(bqm), bqm)
solution = main.run(init_state).result()

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-34da8c1091f1> in <module>()
     10 
     11 init_state = State.from_sample(min_sample(bqm), bqm)
---> 12 solution = main.run(init_state).result()

~/anaconda3/lib/python3.6/concurrent/futures/_base.py in result(self, timeout)
    430                 raise CancelledError()
    431             elif self._state == FINISHED:
--> 432                 return self.__get_result()
    433             else:
    434                 raise TimeoutError()

~/anaconda3/lib/python3.6/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

~/anaconda3/lib/python3.6/concurrent/futures/thread.py in run(self)
     54 
     55         try:
---> 56             result = self.fn(*self.args, **self.kwargs)
     57         except BaseException as exc:
     58             self.future.set_exception(exc)

~/anaconda3/lib/python3.6/site-packages/hybrid/core.py in dispatch(self, future)
    341 
    342         with self.count('dispatch.next'):
--> 343             new_state = self.next(state)
    344 
    345         self.validate_output_state_traits(new_state)

~/anaconda3/lib/python3.6/site-packages/hybrid/flow.py in next(self, state)
    447 
    448         for iterno in range(self.max_iter):
--> 449             state = self.runnable.run(state).result()
    450             state_quality = self.key(state)
    451 

~/anaconda3/lib/python3.6/concurrent/futures/_base.py in result(self, timeout)
    423                 raise CancelledError()
    424             elif self._state == FINISHED:
--> 425                 return self.__get_result()
    426 
    427             self._condition.wait(timeout)

~/anaconda3/lib/python3.6/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

~/anaconda3/lib/python3.6/concurrent/futures/thread.py in run(self)
     54 
     55         try:
---> 56             result = self.fn(*self.args, **self.kwargs)
     57         except BaseException as exc:
     58             self.future.set_exception(exc)

~/anaconda3/lib/python3.6/site-packages/hybrid/core.py in dispatch(self, future)
    341 
    342         with self.count('dispatch.next'):
--> 343             new_state = self.next(state)
    344 
    345         self.validate_output_state_traits(new_state)

~/anaconda3/lib/python3.6/site-packages/hybrid/flow.py in next(self, state)
    134         for component in self.components:
    135             state = component.run(state, defer=False)
--> 136         return state.result()
    137 
    138     def error(self, exc):

~/anaconda3/lib/python3.6/concurrent/futures/_base.py in result(self, timeout)
    423                 raise CancelledError()
    424             elif self._state == FINISHED:
--> 425                 return self.__get_result()
    426 
    427             self._condition.wait(timeout)

~/anaconda3/lib/python3.6/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

~/anaconda3/lib/python3.6/site-packages/hybrid/core.py in submit(self, fn, *args, **kwargs)
     41         # customizable underlying executor (e.g. thread/process/celery/network)
     42         try:
---> 43             return Present(result=fn(*args, **kwargs))
     44         except Exception as exc:
     45             return Present(exception=exc)

~/anaconda3/lib/python3.6/site-packages/hybrid/core.py in dispatch(self, future)
    331             except Exception as exc:
    332                 with self.count('dispatch.resolve.error'):
--> 333                     return self.error(exc)
    334 
    335         if not getattr(self, '_initialized', False):

~/anaconda3/lib/python3.6/site-packages/hybrid/core.py in error(self, exc)
    312         must be explicitly silenced.
    313         """
--> 314         raise exc
    315 
    316     def dispatch(self, future):

~/anaconda3/lib/python3.6/site-packages/hybrid/core.py in dispatch(self, future)
    328         with self.count('dispatch.resolve'):
    329             try:
--> 330                 state = future.result()
    331             except Exception as exc:
    332                 with self.count('dispatch.resolve.error'):

~/anaconda3/lib/python3.6/concurrent/futures/_base.py in result(self, timeout)
    423                 raise CancelledError()
    424             elif self._state == FINISHED:
--> 425                 return self.__get_result()
    426 
    427             self._condition.wait(timeout)

~/anaconda3/lib/python3.6/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

~/anaconda3/lib/python3.6/site-packages/hybrid/core.py in submit(self, fn, *args, **kwargs)
     41         # customizable underlying executor (e.g. thread/process/celery/network)
     42         try:
---> 43             return Present(result=fn(*args, **kwargs))
     44         except Exception as exc:
     45             return Present(exception=exc)

~/anaconda3/lib/python3.6/site-packages/hybrid/core.py in dispatch(self, future)
    341 
    342         with self.count('dispatch.next'):
--> 343             new_state = self.next(state)
    344 
    345         self.validate_output_state_traits(new_state)

~/anaconda3/lib/python3.6/site-packages/hybrid/samplers.py in next(self, state)
    100 
    101     def next(self, state):
--> 102         response = self.sampler.sample(state.subproblem, num_reads=self.num_reads)
    103         return state.updated(subsamples=response)
    104 

~/anaconda3/lib/python3.6/site-packages/dimod/core/sampler.py in sample(self, bqm, **parameters)
    191         elif bqm.vartype is Vartype.BINARY:
    192             h, J, offset = bqm.to_ising()
--> 193             response = self.sample_ising(h, J, **parameters)
    194             response.change_vartype(Vartype.BINARY, energy_offset=offset)
    195             return response

~/anaconda3/lib/python3.6/site-packages/dwave/system/samplers/dwave_sampler.py in sample_ising(self, h, J, **kwargs)
    283         num_variables = len(active_variables)
    284 
--> 285         future = self.solver.sample_ising(h, J, **kwargs)
    286 
    287         return dimod.Response.from_future(future, _result_to_response_hook(active_variables, dimod.SPIN))

~/anaconda3/lib/python3.6/site-packages/dwave/cloud/solver.py in sample_ising(self, linear, quadratic, **params)
    196         # Our linear and quadratic objective terms are already separated in an
    197         # ising model so we can just directly call `_sample`.
--> 198         return self._sample('ising', linear, quadratic, params)
    199 
    200     def sample_qubo(self, qubo, **params):

~/anaconda3/lib/python3.6/site-packages/dwave/cloud/solver.py in _sample(self, type_, linear, quadratic, params)
    251         # Check the problem
    252         if not self.check_problem(linear, quadratic):
--> 253             raise ValueError("Problem graph incompatible with solver.")
    254 
    255         # Mix the new parameters with the default parameters

ValueError: Problem graph incompatible with solver.


```

**Comment by randomir** [comment]
The qpusampler you specified is not compatible with the BQM’s graph. Try either:
- omitting qpu_sampler, or
- using EmbeddingComposite(DWaveSampler()) for the qpu sampler

**Comment by evanlai33** [comment]
Tried with EmbeddingComposite, ran through but returned all zeros results. Thank you!
```Solution: sample=Sample(sample={'x': 0, 'y': 0, 'z': 0, 'a': 0, 'b': 0, 'c': 0}, energy=-1.0, num_occurrences=1)```

**Comment by randomir** [comment]
But it makes sense to assume “auto embedding sampler” will handle the embedding. So the naming, or existence, of qpu_sampler argument is inconsistent with the class name. Its purpose was debugging and testing, but we’ll fix this inconsistency. Thanks @evanlai33!

**Comment by randomir** [comment]
@evanlai33, FWIW, we fixed this in #77 (6d1bce2) by making QPU auto embedding sampler adaptively un-structure child samplers.
